config/backend-config/backend-config.go

```diff
@@ -388,23 +383,53 @@ func (bc *CommonBackendConfig) WaitForConfig(ctx context.Context) error {
 	return nil
 }
 
-// Setup backend config
-func Setup(configEnvHandler types.ConfigEnvI) {
-	if isMultiWorkspace && misc.IsMultiTenant() {
-		backendConfig = new(MultiTenantWorkspaceConfig)
-		backendConfig.(*MultiTenantWorkspaceConfig).CommonBackendConfig.configEnvHandler = configEnvHandler
-	} else if isMultiWorkspace {
-		backendConfig = new(MultiWorkspaceConfig)
-	} else {
-		backendConfig = new(WorkspaceConfig)
-		backendConfig.(*WorkspaceConfig).CommonBackendConfig.configEnvHandler = configEnvHandler
+func NewForDeployment(deploymentType deployment.Type, configEnvHandler types.ConfigEnvI) (BackendConfig, error) {
```

Comment:
Looks like this function is not being used elsewhere. Let us not expose it

Reply:
done